doc/kernel/format_specifications.rdoc

```diff
@@ -0,0 +1,230 @@
+  See {Specifier s}[rdoc-ref:doc/kernel/format_specifications.rdoc@Specifier+s].
+- <tt>%</tt>: Convert a single percent character.
+  See {Specifier %}[rdoc-ref:doc/kernel/format_specifications.rdoc@Specifier+-25].
+
+=== Flags
+
+- Flag <tt><space></tt>: Insert a space before a non-negative number:
+
+    sprintf('%d', 10)  # => "10"
+    sprintf('% d', 10) # => " 10"
+
+  Insert a minus sign for negative value:
+
+    sprintf('%d', -10)  # => "-10"
+    sprintf('% d', -10) # => "-10"
+
+- Flag <tt>#</tt>: Use an alternate format; varies among types:
+
+    sprintf('%x', 100)  # => "64"
+    sprintf('%#x', 100) # => "0x64"
+
+- Flag <tt>+</tt>: Add a leading plus sign for a non-negative number:
+
+    sprintf('%x', 100)  # => "64"
+    sprintf('%+x', 100) # => "+64"
+
+- Flag <tt>-</tt>: Left justify the value in its field:
+
+    sprintf('%6d', 100)  # => "   100"
+    sprintf('%-6d', 100) # => "100   "
+
+- Flag <tt>0</tt>: Buffer with zeros instead of spaces:
+
+    sprintf('%6d', 100)  # => "   100"
+    sprintf('%06d', 100) # => "000100"
+
+- Flag <tt>*</tt>: Use the next argument as the field width:
+
+    sprintf('%d', 20, 14)  # => "20"
+    sprintf('%*d', 20, 14) # => "                  14"
+
+- Flag <tt><n>$</tt>: Convert the (1-based) +n+th argument into this field:
+
+    sprintf("%s %s", 'world', 'hello')     # => "world hello"
+    sprintf("%2$s %1$s", 'world', 'hello') # => "hello world"
+
+=== Width Specifiers
+
+In general, a width specifier determines the width (in characters)
+of the formatted field:
+
+  sprintf('%10d', 100)  # => "       100"
+  # Left-justify if negative.
+  sprintf('%-10d', 100) # => "100       "
+  # Ignore if too small.
+  sprintf('%1d', 100)   # => "100"
+
+=== Type Specifier Details and Examples
+
+==== Specifiers +a+ and +A+
+
+==== Specifiers +b+ and +B+
+
+Convert +argument+ as a binary integer:
+
+The two specifiers +b+ and +B+ behave identically
+except when +0b+ or +0B+ is prefixed.
+
+Flags (all are allowed):
+
+- No flag:
+
+    sprintf('%b', 1)  # => "1"
+    sprintf('%b', 4)  # => "100"
+    # Prefix '..' for negative value.
+    sprintf('%b', -4) # => "..100"
+
+- Flag <tt>#</tt>: Prefix +0b+ or +0B+:
+
+    sprintf('%#b', 1)  # => "0b1"
+    sprintf('%#B', 1)  # => "Bb1"
+    sprintf('%#b', 4)  # => "0b100"
+    # Embed '..' for negative value.
+    sprintf('%#b', -4) # => "0b..100"
+
+- Space flag: Prefix space for non-negative value,
+  <tt>-</tt> for negative value:
+
+    sprintf('% b', 0)  # => " 0"
+    sprintf('% b', 4)  # => " 100"
+    # Prefix '-' for negative value.
+    sprintf('% b', -4) # => "-100"
+
+- Flag <tt>+</tt>: Prefix <tt>+</tt> for non-negative value.
+  <tt>-</tt> for negative value:
+
+    sprintf('%+b', 0)  # => "+0"
+    sprintf('%+b', 4)  # => "+100"
+    sprintf('%+b', -4) # => "-100"
+
+- Flag <tt>-</tt>: Allowed, but has no effect.
+
+- Flag +0+: Allowed, but has no effect.
+
+- Flag <tt>*</tt>: Treat first-following argument as a width specifier
```

Comment:
As I said, "One of the aims here is to make the discussion of a type specifier, say %b, sufficient in itself, without the reader having to carry a lot of info from other sections."  A reader trying to look up %b behavior should not have to read the whole page to learn about flags, width, and precision.

Reply:
Duplicating repetitive content in every section is much worse than documenting shared behavior in one place, and specific exceptions for each specifier, IMO.  If you want to have a bunch of long sections that mostly repeat the same information for different types, you'll have to find another reviewer, because I cannot approve that approach.

Reply:
> Duplicating repetitive content in every section is much worse than documenting shared behavior in one place, and specific exceptions for each specifier, IMO. If you want to have a bunch of long sections that mostly repeat the same information for different types, you'll have to find another reviewer, because I cannot approve that approach.

Ditch the repetitive text, but keep the examples?

Reply:
Have a single example section for each type specifier section showing basic behavior without flags. Beyond that, only keep examples for flags that show type specifier-specific behavior, and those should have text explaining the different behavior.  As I mentioned, for `b` and `B`, the `..` handling for negative numbers and `#` for changing capitalization are exceptions that should be mentioned.  There may be other exceptions, but I think only exceptions should have documentation and examples.  Any flag behavior that is the same across type specifiers does not need explanations or examples in each type specifier section.

Reply:
@jeremyevans, we're very far apart on how to approach this topic.  With your permission, I'll close this PR.

Reply:
Sure, that's fine.